<!doctype html><html><head><link href="rule.css" rel="stylesheet" type="text/css" /></head><body><h1><big>Multiple variables should not be declared on the same line</big> (squid:S1659)</h1><div class="rule-desc"><p>Declaring multiple variable on one line is difficult to read.</p>
<h2>Noncompliant Code Example</h2>

<pre>
class MyClass {

  private int a, b;

  public void method(){
    int c; int d;
  }
}
</pre>
<h2>Compliant Solution</h2>

<pre>
class MyClass {

  private int a;
  private int b;

  public void method(){
    int c;
    int d;
  }
}
</pre></div></body></html>